services/provider/server/server.go

```diff
@@ -1229,13 +1230,39 @@ func (s *OCSProviderServer) getDesiredCephConnection(
 		return nil, fmt.Errorf("failed to extract monitor IPs from configmap %s: %v", monConfigMap, err)
 	}
 
+	var readAffinity *csiopv1a1.ReadAffinitySpec = nil
+	readAffinityOptions := util.GetReadAffinityOptions(storageCluster)
+
+	readaffinityDisabledAnnotation := consumer.Annotations["ocs.openshift.io/disable-read-affinity"] == "true"
+	defaultCrushLocationLabels := []string{
+		"topology.rook.io/chassis",
+		"topology.rook.io/rack",
+		"topology.rook.io/row",
+		"topology.rook.io/pdu",
+		"topology.rook.io/pod",
+		"topology.rook.io/room",
+		"topology.rook.io/datacenter",
+		"topology.kubernetes.io/region",
+		"topology.kubernetes.io/zone",
+		"kubernetes.io/hostname",
+	}
+	var mergedLabels []string
+
+	if readAffinityOptions.Enabled && !readaffinityDisabledAnnotation {
+		mergedLabels = slices.Concat(defaultCrushLocationLabels, readAffinityOptions.CrushLocationLabels)
+		readAffinity = &csiopv1a1.ReadAffinitySpec{
+			CrushLocationLabels: mergedLabels,
+		}
+	}
```

Comment:
This entire code does not take into account duplications. 
Please use a set (`map[string]struct{}`) for the default values, add the additional values to the set, then use `maps.keys` to get back the list without duplicates

Reply:
@Rakshith-R I think we can send duplicates, but lets be sure,

I think we can use slice.compact(s1, s2) too

Reply:
> @Rakshith-R I think we can send duplicates, but lets be sure,

It is handled at cephcsi here 
https://github.com/ceph/ceph-csi/blob/dd93de0bfc8165b8676c6e5045477290077fbf85/internal/util/crushlocation.go#L43-L46